### ISSUE TITLE
Slow typing after upgrade to release-5.23

### ISSUE THREAD

**Issue by jon88** [mattermost/mattermost-server#14755]
#### Summary
After upgrade to release-5.23 typing and repeated keystrokes are very slow compared to relase-5.21 in mattermost-webapp under firefox 76.0.1

#### Steps to reproduce
* mattermost/mattermost-team-edition:release-5.23 and mattermost/mattermost-prod-db:5.23.0
  * watch CPU utilization via htop / top orsm
  * Open mattermost webapp and hold down keystroke as when typing a message
  * Expierence characters printed very slowly
   * See that 1 CPU core is utilized 100% or very high
* mattermost/mattermost-team-edition:release-5.21 and mattermost/mattermost-prod-db:5.23.0
  * watch CPU utilization via htop / top orsm
  * Open mattermost webapp and hold down keystroke as when typing a message
  * Expierence characters printed normally
  * See that no CPU cores are highly utilized

This also happens with mattermost-team-edition:release-5.24 (release-5.24.83.14cd748)
Tested under ubuntu 20.04 with firefox 76.0.1
edit: This seems to be regression of release-5.22 as issue is present in release 5.22

#### Expected behavior
Keystrokes in newer version are at least as fast as in previous release and comparable to character speed when typing in terminal.

#### Observed behavior (that appears unintentional) 
High CPU utilization

**Comment by amyblais** [comment]
@jon88 Can you help try clear your cache / delete your AppData folder for the issue to be resolved?

There was a bug in one of our libraries and we've applied a fix for it.

**Comment by jon88** [comment]
Test with disabling cache in firefox inspect mode under network tab.

| test nr | version   | cache in FF | repeated keysrokes in 10s   | notes |
|:----: | :----------: | :---------: | :-------------------------: | :---------------: |
| 1| release-5.23 | disabled    | 199                         |  |
| 2 | release-5.23 | enabled     | 136                         |  |
| 3 | release-5.23 | enabled     | 192                         | Cleared MM cookies, did a relogin and cleared globally FF cache |
| 4 | release-5.23 | enabled     | 100                         | Waited around 10 minutes after test 3 |
| 5 | release-5.21 | enabled     | 228                         |  |

In my conclusion clearing cache seems to help. The performance is still not comparable to release-5.21 and for some reason after some time it goes back to being slow. Either I did not manage to clear cache correctly (for which I followed: https://support.mozilla.org/en-US/kb/how-clear-firefox-cache) or the broken library gets cached again from somewhere.

**Comment by sudheerDev** [comment]
@jon88 Can you let us know if the issue persists? 
Do you see any console errors when this happens?

**Comment by jon88** [comment]
Clearing cache globally didn't help and even with network/disable Cache to force redownload of all js etc it still feels slower and it seems that it starts becoming slower and slower as with previous release no such problems. I can try to do a fresh install of those different versions to exclude some migration problems.

I can see such errors on console:
```
downloadable font: kern: Too large subtable (font-family: "Open Sans" style:normal weight:600 stretch:100 src index:3) source: https://mattermost.starman.ee/static/files/aa22ff6fd92ecdf402f09a8180354751.woff2
downloadable font: Table discarded (font-family: "Open Sans" style:normal weight:600 stretch:100 src index:3) source: https://mattermost.starman.ee/static/files/aa22ff6fd92ecdf402f09a8180354751.woff2
downloadable font: kern: Too large subtable (font-family: "Open Sans" style:italic weight:400 stretch:100 src index:3) source: https://mattermost.starman.ee/static/files/f01a4402103c9ef9350607c0f739b49d.woff2
downloadable font: Table discarded (font-family: "Open Sans" style:italic weight:400 stretch:100 src index:3) source: https://mattermost.starman.ee/static/files/f01a4402103c9ef9350607c0f739b49d.woff2 
```

**Comment by jon88** [comment]
I did a separate fresh install of mattermost/mattermost-team-edition:release-5.23 with same (as similar as possible) configuration and repeated characters speed in 10s is 230.

Same errors appear on console and served `js` files have the same names.

I then noticed that I have `Style sheet could not be loaded` error on top of the inspect tab in firefox (this appears only in the server we have problems; the fresh install doesn't have such warning). Although exactly the same css files have transfered, with same sizes and 200 responses. I don't know how a missing CSS could be responsible for such error.

**Comment by sudheerDev** [comment]
Can you check if you can replicate the issue in the private browsing tab if you have any browser plugins installed? 

And if you still are seeing issue, we could use a performance profile when this happens for us to debug this better.

To record a performance profile. Right-click in the tab > inspect element > click on performance tab > click on start recording performance and start typing in the channel.

<img width="1479" alt="Untitled" src="https://user-images.githubusercontent.com/4973621/83918007-8154b780-a795-11ea-9075-261cd9618773.png">

Once the actions are performed you can stop recording the profile and hit save 

<img width="808" alt="Untitled1" src="https://user-images.githubusercontent.com/4973621/83918035-8e71a680-a795-11ea-8795-b26ca49458c2.png">

**Comment by jon88** [comment]
Issue is present in private mode as-well. 
As far as I can understand the performance graph then when running normally fps is around 60 fps but when hitting a keystroke then it drops down to 8.26 fps.

See attached: 
[ff-profile.json.gz](https://github.com/mattermost/mattermost-server/files/4745402/ff-profile.json.gz)

**Comment by sudheerDev** [comment]
@jon88 Hey, thanks for the profile, I am looking into it. Firefox dev tools for performance debugging are not great. It makes the work harder than it should be. Could you also check on chrome and if you are seeing the same issues, can you share a performance profile from chrome dev tools? The steps above are pretty much similar.

**Comment by jon88** [comment]
I tested with a windows machine with FF ( version 77) and Chrome: everything worked fine. Chrome was even super-fast.

I then tried with a docker-container firefox on my computer and the problem persists.
I took my laptop which has the same distro/ kernel/ ff:
  - with release-5.23 minFps drops to 10 fps
  - with release-5.21 minFps drops to 35 fps

Same thing is observed by another user who is running: openSUSE Leap 15.1 / ff 76.0.1 (64-bit)

**Comment by lieut-data** [comment]
@jon88 thanks for the specifics, and the good news that it's running fast in at least a subset of environments. 

I admit to not having experience running Firefox in a Docker container. Would I be correct in saying that all the performance issues you've observed are in a Linux environment and running Firefox? Would you be open to testing with Chromium for Linux and seeing if the problem persists there?

**Comment by jon88** [comment]
By default I don't run firefox in a container either but just wanted as-clean state I can provide.
Tested with chrome on linux and seemed to run fine.

It seems that channel where the writing is done affects this issue:
direct message to specific person (who also has this problem): 112 c / 10s
direct message to myself: 213 c / 10s
almost empty private channel: 228 c / 10s

This must be somehow related to #14780 but indeed symptoms don't seem to match.

I don't want to rule out the possibility that different platforms would be slower if they would be used more frequently but I did the same test above with chrome and speed was almost constast 230 c /10s
